src/domain-services/flow-object/flow-object-service.ts

```diff
@@ -0,0 +1,150 @@
+import { type Database } from '@unocha/hpc-api-core/src/db';
+import { type FlowId } from '@unocha/hpc-api-core/src/db/models/flow';
+import {
+  Op,
+  type Condition,
+} from '@unocha/hpc-api-core/src/db/util/conditions';
+import { type OrderByCond } from '@unocha/hpc-api-core/src/db/util/raw-model';
+import type {
+  FieldsOfModel,
+  InstanceOfModel,
+} from '@unocha/hpc-api-core/src/db/util/types';
+import { createBrandedValue } from '@unocha/hpc-api-core/src/util/types';
+import { Service } from 'typedi';
+import { type UniqueFlowEntity } from '../flows/model';
+import { buildSearchFlowsObjectConditions } from '../flows/strategy/impl/utils';
+import { type FlowObjectFilterGrouped } from './model';
+import { buildWhereConditionsForFlowObjectFilters } from './utils';
+
+// Local types definition to increase readability
+type FlowObjectModel = Database['flowObject'];
+type FlowObjectInstance = InstanceOfModel<FlowObjectModel>;
+export type FlowObjectsFieldsDefinition = FieldsOfModel<FlowObjectModel>;
+export type FlowObjectOrderByCond = OrderByCond<FlowObjectsFieldsDefinition>;
+export type FlowObjectWhere = Condition<FlowObjectInstance>;
+@Service()
+export class FlowObjectService {
+  // Merge with getFlowsObjectsByFlows
+  async getFlowIdsFromFlowObjects(
+    models: Database,
+    where: FlowObjectWhere
+  ): Promise<FlowId[]> {
+    const flowObjects = await models.flowObject.find({
+      where,
+    });
+    // Keep only not duplicated flowIDs
+    return [...new Set(flowObjects.map((flowObject) => flowObject.flowID))];
+  }
+
+  async getFlowFromFlowObjects(
+    models: Database,
+    where: FlowObjectWhere
+  ): Promise<UniqueFlowEntity[]> {
+    const flowObjects = await models.flowObject.find({
+      where,
+    });
+    // Keep only not duplicated flowIDs
+    return [
+      ...new Set(
+        flowObjects.map((flowObject) => {
+          return {
+            id: createBrandedValue(flowObject.flowID),
+            versionID: flowObject.versionID,
+          };
+        })
+      ),
+    ];
+  }
+
+  async getFlowObjectByFlowId(models: Database, flowIds: FlowId[]) {
+    return await models.flowObject.find({
+      where: {
+        flowID: {
+          [Op.IN]: flowIds,
+        },
+      },
+    });
+  }
+
+  async getFlowObjectsByFlowObjectConditions(
+    models: Database,
+    flowObjectFilterGrouped: FlowObjectFilterGrouped
+  ): Promise<FlowObjectInstance[]> {
+    const whereClause = buildWhereConditionsForFlowObjectFilters(
+      flowObjectFilterGrouped
+    );
+
+    return await models.flowObject.find({ where: whereClause });
+  }
+
+  async getFlowsObjectsByFlows(
+    models: Database,
+    whereClauses: FlowObjectWhere,
+    orderBy?: FlowObjectOrderByCond
+  ): Promise<FlowObjectInstance[]> {
+    const distinctColumns = [
```

Comment:
Let's declare types instead of forcing them.
```suggestion
    const distinctColumns: Array<keyof FlowObjectInstance> = [
      'flowID',
      'versionID',
    ];
```